### ISSUE TITLE
ADE key location on android

### ISSUE THREAD

**Issue by event** [apprenticeharper/DeDRM_tools#952]
This isn't really an issue. I just found an activation.xml on ADE for Android and wanted to share. It may be worth adding it to FAQ. I used it to dedrm epubs downloaded through Android ADE (I ain't got macosx or windows).
File itself is in /data/data/com.adobe.digitaleditions/files/.adept/activation.xml (root required to copy it). To get key extract /activationInfo/activationServiceInfo/credentials/privateLicenseKey, de-base64 and chop off first 26 bytes (I used "tail -c +27")

**Comment by ElleKayEm** [comment]
Thanks for the info.  Always good to have options.  I'd guess this app could handle the hardened Adobe DRM.  So you may risk getting DRM you can't remove if you use it to open the initial .acsm file.

**Comment by event** [comment]
What is this "hardened DRM" thing? Google says it's been introduced ~5 years ago, but no one goes into much details. Is it hardware-backed with e.g. widevine or something?

**Comment by ElleKayEm** [comment]
No idea how it is "hardened" compared to "regular" Adobe DRM.  All I know is that Adobe Digital Editions 2.0.1 for PC and Mac cannot download epubs with the newer DRM.  So an ebookstore might be more likely to refund because they could see you hadn't actually downloaded a book.

Here's Adobe's page about it:
https://www.adobe.com/ca/solutions/ebook/content-server/faq.html

Adobe Content Server 4.x delivers the DRM these tools can handle.  ACS 5 and later is capable of delivering DRM that cannot be removed by these (or as far as I know, any other) tools depending on whether the ebookstore has made it mandatory or not.  (Not to be confused with ADE version numbers.  ADE 2.0.1 can't handle ACS 5 and later.  ADE 3 and later can.)

**Comment by redthing1** [comment]
This works for me and is super helpful! Able to view books from archive.org

**Comment by TheRealIndru** [comment]
> To get key extract /activationInfo/activationServiceInfo/credentials/privateLicenseKey, de-base64 and chop off first 26 bytes (I used "tail -c +27")

Mine is lacking privateLicenseKey field. Any hints?

**Comment by ElleKayEm** [comment]
You have it authorized with your Adobe ID?

**Comment by TheRealIndru** [comment]
> You have it authorized with your Adobe ID?

Yes, and it lacks that field. Has a certificate field and a signature field, alongside my credentials (Adobe ID).
There is no privateLicenseKey though... Maybe something changed since 2019 when this was posted?

**Comment by j-howell** [comment]
I tried this today and it worked for me.

I installed a fresh copy of Adobe Digital Editions for Android from the Play Store (version 1.0 dated August 10, 2020) on a device with root access. I authorized it using an Adobe ID and closed the app. I then extracted /data/data/com.adobe.digitaleditions/files/.adept/activation.xml. It contained adept:privateLicenseKey under activationInfo, adept:credentials with the correct key.

I did notice that the first post in this issue incorrectly states that privateLicenseKey will be found under activationServiceInfo.

**Comment by TheRealIndru** [comment]
Thank you! Yes, it was under adept:credentials. :) I do not know why I couldn't find the string in it initially.

Now on to the last part:

`de-base64 and chop off first 26 bytes (I used "tail -c +27")`

How do I do that? If I use base64decode.org I get random gibberish.

Same with `base64 --decode` in Terminal.

Later edit: Saved the 'gibberish' in .der format, basically command was:

`echo 'insert_key_here' | base64 --decode | tail -c +27 > activation.der`

And then I loaded that in Calibre in the DeDRM plugin settings.

Looks like it decoded successfully, I might test further.

**Comment by OwlJohn** [comment]
Hello everyone,

Can you remove drm using just an android phone, without using windows or mac? Does anyone know a guide to do it? 

Thank you very much in advance.

**Comment by ElleKayEm** [comment]
Are you just referring to getting the books?  Still using DeDRM for the actual DRM removal? And books from where?

**Comment by OwlJohn** [comment]
> Are you just referring to getting the books? Still using DeDRM for the actual DRM removal? And books from where?

Thanks for your reply. No, I am not using any program. I just wanted to know if it is possible to remove drm from an acsm file using an Android phone. Apprentice Alf does not support Android, I think. Do you know any way to do it? Thank you.

**Comment by ElleKayEm** [comment]
No, I'm not aware of any Android DRM remover for Adobe DRM.

**Comment by OwlJohn** [comment]
> No, I'm not aware of any Android DRM remover for Adobe DRM.

OK. Thank you.
